### DISCUSSION TITLE
Cannot make HyperPixel 4.0 work with Raspberry Pi 3B+ // screen is always black

### DISCUSSION THREAD

**Discussion by fichone** [FundamentalFrequency/.github#71] [General]
Hey guys,

I was really excited by this project and started to gather all the components. I am now to begin assembling PCB, switches & encoders, but display came yesterday, so I wanted to check if everything is working fine as I already returned one, which was cheaper alternative and not working at all. So what I did until now:

- installed Raspberry Pi OS 64 bit version on a SD card
- run the one-line installer as described here: https://shop.pimoroni.com/products/hyperpixel-4?variant=12569485443155
- configuring the screen in /boot/config.txt

After reboot, OS did not load - got stucked on the color screen. It seems like display is turning on, but stays black. Also before configuring the screen, I was able to move the cursor of the mouse with the touch screen.

So, not sure what is wrong here and why I am not able to set it up properly - also not sure that display is working on 100% otherwise I have to return it. Or maybe I am missing something...?

Will appreciate some help here.

regards, 
fichone

**Comment by stonepreston** [comment]
The instructions [here](https://github.com/pimoroni/hyperpixel4/tree/master) seem to be relatively up to date. I would run the curl command listed [here](https://github.com/pimoroni/hyperpixel4/tree/master#installing--uninstalling) and follow the prompts to select your combination of screen and device as described there. 

It seems rotation is set differently on a pi 3 vs pi 4. The section [here](https://github.com/pimoroni/hyperpixel4/tree/master#180-degree-rotation-on-pi-3) describes how to get rotation working on a 3. 

If none of that works there are also instructions for manual installation  [here](https://github.com/pimoroni/hyperpixel4/tree/pi3). It seems those drivers mention they support Raspbian buster, so you might need to install that instead of the latest bullseye release. The buster image can be downloaded from [this page](https://www.raspberrypi.com/software/operating-systems/). Scroll all the way down to the bottom to the legacy image section.

**Comment by fichone** [comment]
Thank you very much for you assistance - it is working now.

Installing only the drivers did not solve the problem, but an old version of Rasp OS (in my case 32 bit Buster as you suggest above) + manual steps of HyperPixel 4.0" Drivers for Raspberry Pi 3B+ from [here](https://github.com/pimoroni/hyperpixel4/tree/pi3) made it work.

Once again - appreciate the support.

**Reply by Mlkurta** [reply]
Im having the same issue on a Pi 3b, but after installing the 32 bit Buster OS I was unable to run the LMN-3 executable. Were you able to get the program to run?

**Reply by stonepreston** [reply]
are you sure you are using the correct build of the binary? Since you are running the 32 bit version of the OS, you will need the arm version of the LMN-3 executable (which is compiled for 32 bit systems), not the aarch64 version (which is for 64 bit).

The link for the most recent 32 bit version of the LMN-3 DAW is [here](https://github.com/FundamentalFrequency/LMN-3-DAW/releases/download/v0.1.1/LMN-3-arm-linux-gnueabihf.zip)
